apps/meteor/client/components/Omnichannel/CustomFieldsForm/index.tsx

```diff
@@ -0,0 +1,67 @@
+import React, { ReactElement } from 'react';
+import { Controller, useFormContext } from 'react-hook-form';
+
+import { CustomSelectInput } from './CustomSelectInput';
+import { CustomTextInput } from './CustomTextInput';
+import { useCustomFields } from './hooks/useCustomFields';
+
+const CustomFieldsForm = (): ReactElement | null => {
+	const result = useCustomFields();
+
+	const { register, control } = useFormContext<{ livechatData: Record<string, unknown> }>();
+
+	const { data, isLoading } = result;
+
+	if (!data || isLoading) {
+		return null;
+	}
+
+	if (result.isError) {
+		throw result.error;
+	}
+
+	return (
+		<>
+			{data.customFields.map((customField) => {
+				// TODO: REMOVE FILTER ONCE THE ENDPOINT SUPPORTS A SCOPE PARAMETER
+				if (customField.scope !== 'room') {
+					return null;
+				}
```

Comment:
You could filter the array before mapping.

Reply:
nice!